desc/input_reader.py

```diff
@@ -845,91 +848,77 @@ def descout_to_input(  # noqa: C901 - fxn too complex
+        iota = grid.compress(eq.compute("iota", grid=grid)["iota"])
+        current = grid.compress(eq.compute("current", grid=grid)["current"])
+
+        pres_profile = PowerSeriesProfile.from_values(
+            rho, pressure, order=10, sym=False
+        ).params
+        iota_profile = PowerSeriesProfile.from_values(
+            rho, iota, order=eq.L, sym=False
+        ).params
+        curr_profile = PowerSeriesProfile.from_values(
+            rho, current, order=eq.L, sym=False
+        ).params
+
+        if eq.iota:
             char = "i"
-            iseven_pres = int(eq0._pressure.basis.sym == "even") + 1
-            iseven_iota = int(eq0._iota.basis.sym == "even") + 1
-            pres_profile = np.zeros((eq0.L + 1,))
-            iota_profile = np.zeros((eq0.L + 1,))
-            pres_profile[: eq0.L + 1 : iseven_pres] = eq0._pressure.params
-            iota_profile[: eq0.L + 1 : iseven_iota] = eq0._iota.params
-
-            idxs = np.linspace(0, eq0.L - 1, eq0.L, dtype=int)
-            for l in idxs:
-                f.write(
-                    "l: {:3d}\tp = {:16.8E}\t{} = {:16.8E}\n".format(
-                        int(l), pres_profile[l], char, iota_profile[l]
-                    )
-                )
+            profile = iota_profile
         else:
-            assert (
-                eq0.pressure.__class__.__name__ == "PowerSeriesProfile"
-                and eq0.current.__class__.__name__ == "PowerSeriesProfile"
-            ), "Equilibrium must have power series profiles for ascii io"
             char = "c"
-            iseven_pres = int(eq0._pressure.basis.sym == "even") + 1
-            iseven_curr = int(eq0._current.basis.sym == "even") + 1
-            pres_profile = np.zeros((eq0.L + 1,))
-            curr_profile = np.zeros((eq0.L + 1,))
-            pres_profile[: eq0.L + 1 : iseven_pres] = eq0._pressure.params
-            curr_profile[: eq0.L + 1 : iseven_curr] = eq0._current.params
-
-            idxs = np.linspace(0, eq0.L - 1, eq0.L, dtype=int)
-            for l in idxs:
-                f.write(
-                    "l: {:3d}\tp = {:16.8E}\t{} = {:16.8E}\n".format(
-                        int(l), pres_profile[l], char, curr_profile[l]
-                    )
-                )
+            profile = curr_profile
 
-        f.write("\n")
+        idxs = np.linspace(0, eq.L - 1, eq.L, dtype=int)
+        for l in idxs:
+            f.write(
+                "l: {:3d}  p = {:15.8E}  {} = {:15.8E}\n".format(
+                    int(l), pres_profile[l], char, profile[l]
+                )
+            )
 
         f.write("\n# fixed-boundary surface shape\n")
+
         # boundary parameters
-        if eq0.sym:
-            for k, (l, m, n) in enumerate(eq0.surface.R_basis.modes):
-                if abs(eq0.Rb_lmn[k]) > 1e-8:
+        if eq.sym:
+            for k, (l, m, n) in enumerate(eq.surface.R_basis.modes):
+                if abs(eq.Rb_lmn[k]) > 1e-8:
```

Comment:
Wait I did not realize, the desc output to input only saves boundary coefficients larger than 1e-8?

I think at the least this threshold should be an input parameter, and I would put it lower to like 1e-10 as the default

Reply:
#696 not for this PR but for the future making this issue

Reply:
Iirc you were the one who added this feature and I recommended making it an option at the time